validation-test/compiler_crashers_2_fixed/0186-rdar46497155.swift

```diff
@@ -20,10 +20,9 @@ struct E {
 
 func foo(arr: [E], other: P) -> Bool {
   return arr.compactMap { i in
-    // expected-error@-1 {{generic parameter 'ElementOfResult' could not be inferred}}
     var flag = false
     return try? i.getB(&flag)
-  }.compactMap { u -> P? in
+  }.compactMap { u -> P? in // expected-error {{nable to infer type of a closure parameter 'u' in the current context}}
```

Comment:
typo "nable" -> "unable" :)

Reply:
Ah, that's okay, I'll fix it later, it's just a pattern match anyway :)